### PR TITLE
MAGIC value for powerReason should be stored in RTC backup on Horus

### PR DESCRIPTION
**Pull request by bsongis**
- [X] Allows the radio main application to start correctly when coming from an app in RAM
- [X] Refactoring of the `UNEXPECTED_SHUTDOWN()` process
- [X] Saves the RTC battery as BACKUP RAM is not needed on normal shutdown
- [X] On PWR OFF on an ALERT, the radio won't show the next alert before shutting down
- [X] Bootloader flashable through the SD Manager

### CODE REVIEW

**Comment by MikeBland** [comment]
I'm suggesting using one of the 20 RTC Backup registers, not a location in the backup RAM.
If the backup RAM is enabled when power is off, the life of the RTC battery is reduced to about half.

File: radio/src/storage/rtc_backup.h [hunk 1]
```diff
@@ -23,15 +23,9 @@
...
-enum PowerReason {
-  SHUTDOWN_REQUEST = 0xDEADBEEF,
-  SOFTRESET_REQUEST = 0xCAFEDEAD,
-};
-
 PACK(struct RamBackup {
-  uint32_t shutdownRequest;
   uint16_t size;
-  uint8_t data[4090];
+  uint8_t data[4096];
```
Comment (raphaelcoeffic) [inline radio/src/storage/rtc_backup.h]: `4K - sizeof(uint16_t) == 4094`?

Comment (bsongis) [reply radio/src/storage/rtc_backup.h]: ouch

File: radio/src/functions.cpp [hunk 2]
```diff
@@ -35,7 +35,7 @@ void testFunc()
 #endif
 
   // for testing the WD reset uncomment the following line
-  // while (1);
+  while (1);
```
Comment (bsongis) [inline radio/src/functions.cpp]: ATTENTION !

File: radio/src/targets/common/arm/stm32/rtc_driver.cpp [hunk 3]
```diff
@@ -67,27 +67,28 @@ void rtcInit()
...
+
   struct gtm utm;
   rtcGetTime(&utm);
   g_rtcTime = gmktime(&utm);
+#endif
 
-#if defined(RAMBACKUP)
+#if defined(RTC_BACKUP_RAM) && !defined(BOOT)
   RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_BKPSRAM, ENABLE);
   PWR_BackupRegulatorCmd(ENABLE);
```
Comment (raphaelcoeffic) [inline radio/src/targets/common/arm/stm32/rtc_driver.cpp:93]: The manual says that `PWR_GetFlagStatus(PWR_FLAG_BRR)` should be checked after the regulator is turned on to make sure the regulator is ready. Is this something we should care about?

Comment (bsongis) [reply radio/src/targets/common/arm/stm32/rtc_driver.cpp:93]: well, we have not need any issue since day 1 on this, I propose to keep it unchanged

**Pull request merged**